content/document.js

```diff
@@ -240,6 +240,9 @@ const read = memoize((folder) => {
     CONTENT_ARCHIVED_ROOT && filePath.startsWith(CONTENT_ARCHIVED_ROOT);
 
   const rawContent = fs.readFileSync(filePath, "utf8");
+  if (!rawContent) {
+    throw new Error(`${filePath} is an empty file`);
+  }
```

Comment:
This was a sneak-in. During the debug-research phase I discovered that weird things can happen further down if there exists an `index.html` file that accidentally exists but is entirely empty. 

What do you think; should we refactor (perhaps as a follow-up) so that instead of doing:
```
if (fs.existsSync(filePath)) {
  ...
  rawContent = fs.readSync(filePath);
  ...
}
``` 
perhaps it's better to combine that into 1 read operation and catch if it can't be read instead of first checking if the file exists. 
I don't think there's any noticeable performance to gain from doing it all in one, but you could get a truthy `fs.existsSync()` but if you don't have permission to it or it's a directory or it's corrupt then you might get confusing errors.

Reply:
I think that makes sense, which line would oyu then catch instead? The `fm` call?

Reply:
I think the topic is off-topic. I think it's often a mistake to do this:
```
if (fs.existsSync(filePath)) {
  return "Yes, the file exists and everything's fine. Please go ahead and assume it's perfect.";
}
```
Because that might lead to false assumptions because it could be that it exists but it's not a file or it's a file but you don't have permission to read it. And it's very possible that it exists but is corrupted and 0 bytes.